flow/scripts/floorplan.tcl

```diff
@@ -87,6 +87,38 @@ remove_buffers
 
 source $::env(SCRIPTS_DIR)/report_metrics.tcl
 report_metrics "floorplan final"
+set num_instances [llength [get_cells -hier *]]
+puts "number instances before restructure is $num_instances"
+puts "Design Area before restructure"
+report_design_area
+
+if { [info exist ::env(SYNTH_HIER_AREA_RECOVER)] && $::env(SYNTH_HIER_AREA_RECOVER) == 1 } {
+
+  write_verilog $::env(RESULTS_DIR)/2_pre_abc.v
+
+  set tielo_cell_name [lindex $env(TIELO_CELL_AND_PORT) 0]
+  set tielo_lib_name [get_name [get_property [get_lib_cell $tielo_cell_name] library]]
+  set tielo_port $tielo_lib_name/$tielo_cell_name/[lindex $env(TIELO_CELL_AND_PORT) 1]
+
+  set tiehi_cell_name [lindex $env(TIEHI_CELL_AND_PORT) 0]
+  set tiehi_lib_name [get_name [get_property [get_lib_cell $tiehi_cell_name] library]]
+  set tiehi_port $tiehi_lib_name/$tiehi_cell_name/[lindex $env(TIEHI_CELL_AND_PORT) 1]
+
+  restructure -liberty_file $::env(DONT_USE_SC_LIB) -target "area" \
+        -tiehi_port $tiehi_port \
+        -tielo_port $tielo_port \
+        -work_dir $::env(RESULTS_DIR)
+
+  # remove buffers inserted by abc
+  remove_buffers
+
+  write_verilog $::env(RESULTS_DIR)/2_post_abc.v
+}
+
+set num_instances [llength [get_cells -hier *]]
+puts "number instances after restructure is $num_instances"
+puts "Design Area after restructure"
+report_design_area
```

Comment:
This could become: `report_metics "floorplan post-restructure"` or whatever the title you need it to be.

Reply:
Ok, I'll try this and see if all metrics comes out as desired.

Reply:
It looks like quite a baggage to include. Since this is optional feature, don't want to bombard with too many metrics. As feature matures and becomes default, we can include full metrics.

Reply:
same - move inside the if condition